lib/barbeque/execution_poller.rb

```diff
@@ -0,0 +1,26 @@
+module Barbeque
+  class ExecutionPoller
+    def run
+      Barbeque::JobExecution.running.find_in_batches do |job_executions|
+        job_executions.shuffle.each do |job_execution|
```

Comment:
Nice catch. I've added early return https://github.com/cookpad/barbeque/pull/38/commits/1425fe772ff6ac1e45b21f8400e19f04cad1e094 and set shorter network timeout (default is 60s, which is too long for barbeque-worker) https://github.com/cookpad/barbeque/pull/38/commits/cc6647cee361e0011d97073c78124b73d31b73ad .

> it's unlikely for S3 in your region

😂